pkg/genericapiserver/genericapiserver.go

```diff
@@ -348,19 +411,54 @@ func (s *GenericAPIServer) Run(options *options.ServerRunOptions) {
 		MaxHeaderBytes: 1 << 20,
 	}
 
+	insecureStartedCh := make(chan struct{})
 	glog.Infof("Serving insecurely on %s", insecureLocation)
 	go func() {
 		defer utilruntime.HandleCrash()
+
+		startedOnce := false
 		for {
 			if err := http.ListenAndServe(); err != nil {
 				glog.Errorf("Unable to listen for insecure (%v); will try again.", err)
+			} else if !startedOnce {
+				startedOnce = true
+				close(secureStartedCh)
 			}
 			time.Sleep(15 * time.Second)
 		}
 	}()
+
+	<-secureStartedCh
+	<-insecureStartedCh
+	s.RunPostStartHooks(PostStartHookContext{})
+
 	select {}
 }
 
+// RunPostStartHooks runs the PostStartHooks for the server
+func (s *GenericAPIServer) RunPostStartHooks(context PostStartHookContext) {
```

Comment:
This comment was about where to put the code to implement the hook logic. genericapiserver.go is big enough already. Let's put the new code into hook.go or even into another subpackage.